### ISSUE TITLE
Revising the requirements

### ISSUE THREAD

**Issue by zeritte** [bounswe/bounswe2020group4#49]
All of us are assigned to that. (including @burakcuhadar) **Deadline: 20.03.2020 Friday@23.59**
Assistants have given feedbacks about our Requirements v1. During the break, we better fix the issues. The ones below are just the ones they have caught, we can fix similar issues. ***Important reminder: some of the issues may have already been fixed since they examined the version on March 8 Sunday, 23:55***  

- [x]  @burakcuhadar In the glossary, what you mean (i.e., “A functionality for customer users that enables them to save the products they want to buy.”) by “cart” is actually “list”. Also, definitions can be written regardfully and glossary can be enlarged. 
- [x] @cagric0  I believe it is better to state the functionalities first, and then the user types in the requirements. Yet, you start by stating the functionalities of the customer to your requirements without defining them. It is an unconventional style and harder to follow. It also causes different issues, where some of them are defined below.  
- [x] @burakcuhadar In 1.1.1.1.6.3, is it “after the items have been shipped” or “after the items have been delivered”?  
- [x] @burakcuhadar About 1.1.1.1.8 “Customers shall be able to comment on products.”, how about the products that are not purchased by the customer? Same applies for 1.1.1.1.9. 
- [x] @egirgin 1.1.1.1.10 “Customers shall be able to access the platform from any software device.” is not clear. Same applies for 1.1.1.2.5, 1.1.1.3.4, and 1.1.1.4.2. Also, with your unconventional style, you repeat the same requirements four times for the four different user types. You should avoid requirement repetition while writing your requirements document just like you avoid code repetition while writing code.  
- [x] @burakcuhadar About 1.1.1.2.2, vendors should be able to provide one or more location (but at least one).  
- [x] @zeritte In 1.1.1.2.3, how about specifying the reason for the cancellation?  
- [x] @burakcuhadar In 1.1.1.2.8 “Vendors can not make purchases using their vendor account.”, you used the word “can”. Instead, you should use either “should” or “shall”.  
- [x] @eridincu Another issue with your unconventional style causes problem for you as well. For example, in 1.1.1.3.3, you state that “Guests shall be able to read user comments about products.”, whereas you did not define such functionality for customers and vendors. However, I am quite sure that they can also read user comments.  
- [x] @olcaytoturker About 1.1.2.1 “Customer users shall message with vendors directly.”, in which conditions? Let me rephrase my question. Can a customer send a message to a vendor just for fun? Can a customer send a message to a vendor after purchasing some of the vendor’s product? Same applies for 1.1.2.2. Clearly elaborate on each requirement.  
- [x] @eylulyalcinkaya Requirements should be as atomic as possible. In 1.1.3.1 “Customer and vendor users shall provide an e-mail and a password while they are signing up. Besides, vendors shall provide their location during registration.”, you state completely different requirements in a single requirement. Also, you are talking about “signing in” under the “Sign up” requirement heading.  
- [x] @burakcuhadar About 1.1.3.2, you used “should”. Yet, I think you should have used “shall”.  
- [x] @berkayalkan In 1.1.3.5, what kind of verification mechanism are you stating?  
- [x] @koraycetiin About 1.1.3.6, how about asking customers to accept an agreement during the checkout after writing the payment information?  
- [x] @berkeozdemir2016400246 In 1.2.1.1.1, what do you mean by “all information”. Clearly elaborate every detail and avoid using words like “such as” in your requirements.  
- [x] @burakcuhadar In 1.2.1.1.2, you mention “semantic search”. Yet, I don’t know what a semantic search mechanism is since it is not defined in your glossary.  
- [x] @katakor In 1.2.1.1.3, what is a “combinative search”?  
- [x] @burakcuhadar In 1.2.1.1.3.1, can you sort products only by their prices? In 1.2.1.1.3.2, why did you use the word “should”? Also, again rather than saying “such as”, clearly elaborate each sortable options in your requirements. Same occurs for 1.2.1.2.5. Moreover, there is another system requirement for “Sorting Mechanism”. If so, why do you state requirements about sorting under “Search Mechanism”?  
- [x]  @burakcuhadar In 1.2.2.1 and 1.2.2.1, which products a customer can comment or rate? The ones that are purchased by the customer or any product? 
- [x] @katakor In 1.2.2.3, under what conditions can customers and admins message with vendor?  
- [x] @burakcuhadar What you mean with 1.2.3.2 “All user types (Guest, Customer, Vendor, Admin) shall have different interfaces.” is not clear.  
- [x] @cagric0  In 1.2.4.1, there is a grammar mistake. 
- [x] @egirgin In 1.2.5.1.1 , how about the details of an active order (i.e., processing stage, cargo stage)? Same applies for 1.2.5.2.1.  
- [x] @zeritte In 1.2.5.1.3, as far as I understand, a customer can cancel his/her order during the cargo stage? Is that right?  
- [x] @eridincu About 1.2.5.1.4, are you going to count the money for canceled and returned orders?  
- [x] @burakcuhadar What did you start e-numeration from 0 for 1.2.5.2.0? Also, there is no functionality for a vendor to accept an order. When a customer makes an order, it is automatically accepted by the system, and directed to the corresponding vendor. If there is an extraordinary problem (e.g., flood in vendor’s storehouse), then vendor can cancel the order by specifying the reason.  
- [x]  @olcaytoturker Vendors should specify their stock information while adding a product to the platform. I cannot see any requirement related to that. 
- [x] @eylulyalcinkaya In 1.2.8.1, I see simple tense and the word “must”. Similar issues apply to 1.2.8.2 and 1.2.8.3 as well. 
- [x] @berkayalkan 1.2.8.3 seems like a user requirement rather than a system requirement.  
- [x] @koraycetiin In 2.1.2, it is not clear what you mean by “PWA”. You should define it in your glossary.  
- [x] @burakcuhadar About 2.1.6, how about English characters? 
- [x] @berkeozdemir2016400246 In 2.1.7, it is not clear what you mean by “communication”.  
- [x] @katakor Non-functional requirements can be enlarged (e.g., performance).

**Comment by egirgin** [comment]
While fixing the "software devices" requirements, I removed them from the user requirements and moved to the 2.1(availability) subtitle. This caused a **change in the indices** of the requirements of 1.1.1.2 Vendor and 1.1.1.3 Guest User. Please take that into account when referring to those requirements. Customer and Admin subsection indices were not affected.

**Comment by koraycetiin** [comment]
To fix the requirement 1.1.3.6, I've added a new requirement 1.2.8.4 because modification was about payment
